### ISSUE TITLE
VH1 no longer downloads videos

### ISSUE THREAD

**Issue by Sipherdrakon** [ytdl-org/youtube-dl#29766]
## Checklist

- [x] I'm reporting a broken site support
- [x] I've verified that I'm running youtube-dl version **2021.06.06**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [x] I've searched the bugtracker for similar issues including closed ones

## Verbose log

```
youtube-dl "https://www.vh1.com/episodes/bxktgs/love-hip-hop-atlanta-a-ruff-road-season-10-ep-5" --verbose
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['https://www.vh1.com/episodes/bxktgs/love-hip-hop-atlanta-a-ruff-road-season-10-ep-5', '--verbose']
[debug] Encodings: locale cp1252, fs utf-8, out utf-8, pref cp1252
[debug] youtube-dl version 2021.06.06
[debug] Python version 3.9.4 (CPython) - Windows-10-10.0.18362-SP0
[debug] exe versions: ffmpeg 4.4-full_build-www.gyan.dev, ffprobe 4.4-full_build-www.gyan.dev
[debug] Proxy map: {}
[vh1.com] bxktgs: Downloading webpage
Traceback (most recent call last):
  File "C:\Program Files\Python39\Scripts\youtube-dl-script.py", line 33, in <module>
    sys.exit(load_entry_point('youtube-dl==2021.6.6', 'console_scripts', 'youtube-dl')())
  File "C:\Program Files\Python39\lib\site-packages\youtube_dl-2021.6.6-py3.9.egg\youtube_dl\__init__.py", line 475, in main
    _real_main(argv)
  File "C:\Program Files\Python39\lib\site-packages\youtube_dl-2021.6.6-py3.9.egg\youtube_dl\__init__.py", line 465, in _real_main
    retcode = ydl.download(all_urls)
  File "C:\Program Files\Python39\lib\site-packages\youtube_dl-2021.6.6-py3.9.egg\youtube_dl\YoutubeDL.py", line 2068, in download
    res = self.extract_info(
  File "C:\Program Files\Python39\lib\site-packages\youtube_dl-2021.6.6-py3.9.egg\youtube_dl\YoutubeDL.py", line 808, in extract_info
    return self.__extract_info(url, ie, download, extra_info, process)
  File "C:\Program Files\Python39\lib\site-packages\youtube_dl-2021.6.6-py3.9.egg\youtube_dl\YoutubeDL.py", line 815, in wrapper
    return func(self, *args, **kwargs)
  File "C:\Program Files\Python39\lib\site-packages\youtube_dl-2021.6.6-py3.9.egg\youtube_dl\YoutubeDL.py", line 836, in __extract_info
    ie_result = ie.extract(url)
  File "C:\Program Files\Python39\lib\site-packages\youtube_dl-2021.6.6-py3.9.egg\youtube_dl\extractor\common.py", line 534, in extract
    ie_result = self._real_extract(url)
  File "C:\Program Files\Python39\lib\site-packages\youtube_dl-2021.6.6-py3.9.egg\youtube_dl\extractor\vh1.py", line 40, in _real_extract
    videos_info = self._get_videos_info(mgid)
  File "C:\Program Files\Python39\lib\site-packages\youtube_dl-2021.6.6-py3.9.egg\youtube_dl\extractor\mtv.py", line 213, in _get_videos_info
    video_id = self._id_from_uri(uri)
  File "C:\Program Files\Python39\lib\site-packages\youtube_dl-2021.6.6-py3.9.egg\youtube_dl\extractor\mtv.py", line 39, in _id_from_uri
    return uri.split(':')[-1]
AttributeError: 'NoneType' object has no attribute 'split'
```

## Description

It seems the mtv extractor is causing the vh1 extractor to throw an error. I don't know enough about python to help fix this but hopefully it isn't too big a fix.

**Comment by dirkf** [comment]
The VH1 extractor didn't succeed in `_extract_triforce_mgid()` and passed the `None` result, causing this error.

`_extract_triforce_mgid()` looks for some JSON being assigned to JS variable `triforceManifestFeed` in the webpage, but obviously it's not there.

**Comment by Sipherdrakon** [comment]
So I tried adding the following to the VH1 extractor because nick.com still works but that didn't work yielding a 500 error
```
    def _get_feed_query(self, uri):
        return {
            'feed': 'vh1_arc_player_prime',
            'mgid': uri,
        }

```

Then I tried adding this extra bit from nick.com as that extractor still works, but then I got ```ValueError: not enough values to unpack (expected 3, got 1)```
```
    def _extract_mgid(self, webpage):
        mgid = self._search_regex(r'"media":{"video":{"config":{"uri":"(mgid:.*?)"', webpage, 'mgid', default=None)
        return mgid

    def _real_extract(self, url):
        domain, video_type, display_id = re.match(self._VALID_URL, url).groups()
        if video_type.startswith("episodes"):
            return super()._real_extract(url)
        video_data = self._download_json(
            'http://%s/data/video.endLevel.json' % domain,
            display_id, query={
                'urlKey': display_id,
            })
        return self._get_videos_info(video_data['player'] + video_data['id'])

```

Sadly I don't think I am much closer to a solution.

**Comment by Sipherdrakon** [comment]
Closed yt-dlp/yt-dlp#783
